client/src/rpc_response.rs

```diff
@@ -220,9 +220,17 @@ pub struct RpcStakeActivation {
     pub inactive: u64,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct RpcTokenAmount {
+    pub amount: f64,
+    pub raw_amount: u64,
```

Comment:
wdyt about this:
```suggestion
    pub ui_amount: f64,
    pub decimals: u8,
    pub amount: String, // "u64",
```

Reply:
UiTokenAccount could then include an `amount: RpcTokenAmount` instead of `amount: u64` too -- that'd allow me to clean up even more token-cli code!

Reply:
Same question, though, about how much of this you want returned from the rpc_client methods?

Reply:
Returning `UiTokenAmount` from rpc_client balance/supply methods would be super.

I feel like we could also start a trend with a `pub type RpcAmount = String` type as well in client/

Reply:
> UiTokenAccount could then include an `amount: RpcTokenAmount` instead of `amount: u64` too -- that'd allow me to clean up even more token-cli code!

This is possible, but requires an additional RPC call everywhere we're using UiTokenAccount to pull the mint and determine the decimals. Could get a little messy in the `getTokenAccountsByOwner` with program id case. Worth it?

Reply:
I was thinking the rpc server could do it.  Here's my current implementation of `solana-token accounts`:
```rust
fn command_accounts(config: &Config, token: Option<Pubkey>) -> CommmandResult {
    let accounts = config
        .rpc_client
        .get_token_accounts_by_owner_with_commitment(
            &config.owner.pubkey(),
            match token {
                Some(token) => TokenAccountsFilter::Mint(token),
                None => TokenAccountsFilter::ProgramId(spl_token::id()),
            },
            config.commitment_config,
        )?
        .value;
    if accounts.is_empty() {
        println!("None");
    }

    println!("Account                                      Token                                        Balance");
    println!("-------------------------------------------------------------------------------------------------");
    for (address, account) in accounts {
        let balance = match config
            .rpc_client
            .get_token_account_balance_with_commitment(&address, config.commitment_config)
        {
            Ok(response) => response.value.to_string(),
            Err(err) => format!("{}", err),
        };

        println!("{:<44} {:<44} {}", address, account.mint, balance);
    }
    Ok(None)
}
```
where I need to call `get_token_account_balance_with_commitment()` because the amounts that I get back from `get_token_accounts_by_owner_with_commitment()` are the raw amounts instead of the ui amounts.

Reply:
Sounds like a yes, then :)
I'd like to do that in a follow-up pr, if that's okay? It will involve a bit of churn to pass account-type-specific data in here: https://github.com/solana-labs/solana/blob/master/account-decoder/src/lib.rs#L48

Reply:
Yes and yes